src/ET_Client.php

```diff
@@ -161,24 +182,29 @@ function __construct($getWSDL = false, $debug = false, $params = null)
 		}		
 		$this->refreshToken();
 
-		try {
-			//$url = $this->baseUrl."/platform/v1/endpoints/soap?access_token=".$this->getAuthToken($this->tenantKey);
-			$url = $this->baseUrl."/platform/v1/endpoints/soap";
-			
-			//$endpointResponse = ET_Util::restGet($url, $this);			
-			$endpointResponse = ET_Util::restGet($url, $this, $this->getAuthToken($this->tenantKey));		
-			//echo "endpoint:  \n";
-			//print_r($endpointResponse);
-							
-			$endpointObject = json_decode($endpointResponse->body);			
-			if ($endpointObject && property_exists($endpointObject,"url")){
-				$this->endpoint = $endpointObject->url;			
-			} else {
-				throw new Exception('Unable to determine stack using /platform/v1/endpoints/:'.$endpointResponse->body);			
-			}
-			} catch (Exception $e) {
-			throw new Exception('Unable to determine stack using /platform/v1/endpoints/: '.$e->getMessage());
-		} 		
+		if ($this->baseSoapUrl) {
+            $this->endpoint = $this->baseSoapUrl;
+		} else {
+            $cache = new ET_CacheService($this->clientId, $this->clientSecret);
```

Comment:
Hello 👋 , why isnt this cache optional? We are having issues with that in production because the cache service has a dependency of the file system